lib/worker_wrapper.js

```diff
@@ -637,4 +648,12 @@ WorkerWrapper.prototype._cancelForceStop = function() {
     }
 };
 
+/**
+ * Adds this worker to master's restart queue
+ * @public
+ */
+WorkerWrapper.prototype.softRestart = function() {
+    this._restartQueue.push(this);
```

Comment:
to reduce the relation between WW & restartQueue and incapsulate queue in the Master, you can implement a method like `master.scheduleWorkerRestart(worker)` and use it, instead of passing the queue around.